wlauto/workloads/youtube/__init__.py

```diff
@@ -0,0 +1,125 @@
+    1. The workload has 4 test scenarios, determined by the ``video_source`` parameter in
+       the agenda file. These are ``home``, ``my_videos``, ``search``, and ``trending``.
+       - A. home - Scrolls down once on the app's home page, then select and plays the video
+            that appears at the top of the list.
+       - B. my_videos - Goes to the 'My Videos' section of the user's account page and plays a
+            video from there. The user must have at least one uploaded video for this to work.
+       - C. search - Goes to the search view and search a specific term, then plays the first
+            video in the search results is played. The parameter ``search_term`` must also be
+            provided in the agenda for this to work.
+       - D. trending - Goes to the 'Trending Videos' section of the app, and plays the first
+            video in the trending videos list.
+       For each scenario, the subsequent test steps are performed.
+    2. Navigating to the general app settings page to disable autoplay.
+    3. Jumping forward in the video by touching the video seek bar and letting the video play
+       for a few seconds.
+    4. Expanding the info card that shows video metadata, then collapsing it again.
+    5. Scrolling down to the end of related videos and comments under the info card, and then
+       back up to the beginning. A maximum of 5 swipe actions is performed in either direction.
+    '''
+
+    parameters = [
+        Parameter('dumpsys_enabled', kind=bool, default=True,
+                  description='''
+                  If ``True``, dumpsys captures will be carried out during the
+                  test run.  The output is piped to log files which are then
+                  pulled from the phone.
+                  '''),
+        Parameter('video_source', kind=str, default='search',
+                  allowed_values=['home', 'my_videos', 'search', 'trending'],
+                  description='''
+                  Determines where to play the video from. This can either be from the
+                  YouTube home, my videos section, trending videos or found in search.
+                  '''),
+        Parameter('search_term', kind=str, default='Big Buck Bunny 60fps 4K - Official Blender Foundation Short Film',
+                  description='''
+                  The search term to use when ``video_source`` is set to ``search``.
+                  Not applicable otherwise.
+                  '''),
+    ]
+
+    view = package + '/com.google.android.apps.youtube.app.WatchWhileActivity'
+
+    instrumentation_log = name + '_instrumentation.log'
+
+    def __init__(self, device, **kwargs):
+        super(Youtube, self).__init__(device, **kwargs)
+        self.run_timeout = 300
+        self.output_file = os.path.join(self.device.working_directory, self.instrumentation_log)
+
+    def validate(self):
+        super(Youtube, self).validate()
+        self.uiauto_params['package'] = self.package
```

Comment:
`validate` is meant just for validating parameters, setting of `uiauto_params` should be done in `initialize`